pkg/autoscaler/algorithms/proportional.go

```diff
@@ -31,15 +31,15 @@ func (a *Proportional) GetDesiredReplicas(metric Metric, replicas int32) int32 {
 	ratio := (metric.Metric.Value / metric.TargetValue)
 	proportional := float64(replicas) * ratio
 	switch metric.TargetType {
-	// Proportional
+	// Proportional, cannot scale to zero
```

Comment:
What about the case where someone wants their HorizontalAutoscaler to scale their resource down all the way, like in the case of a queue? If there's no messages in the queue, I wouldn't want a replica for that, if it was referring to the pods that is.

Reply:
This is the AverageValue case. The math doesn't work (divide by zero) if you try to scale to zero using utilization/value metrics.

Reply:
Ah my bad, I misunderstood where I was at in the code.